src/vsr.zig

```diff
@@ -1917,3 +1918,52 @@ const ViewChangeHeadersArray = struct {
         headers.array.appendAssumeCapacity(Headers.dvc_blank(op));
     }
 };
+
+pub const Op = struct {
+    pub fn checkpoint_before(checkpoint: u64) ?u64 {
+        assert(checkpoint_valid(checkpoint));
```

Comment:
If the namespace becomes `Checkpoint` then we can also use only `assert(valid(checkpoint))` here, which will be nice.